### PR TITLE
Custom Selector - options are not loaded after a non-existing option …

### PR DESCRIPTION
**Pull request by ashklianko**
…is entered into the filter input #3407

-Request is paginated and might have query params, thus caching should not be available

### CODE REVIEW

**Review by alansemenov — CHANGES_REQUESTED** [review]
This fix will break paging and Custom Selector will never load more than 10 items, something that was fixed in https://github.com/enonic/app-contentstudio/issues/3201 (`start` parameter never gets appended to service url).

So while the fix itself is correct, we should find out why the paging got broken in the first place.

To test, use content type called "Color picker" in the Features app that has 16 items in its Custom Selector.

**Comment by ashklianko** [comment]
Updated

File: modules/lib/src/main/resources/assets/js/app/inputtype/customselector/CustomSelectorComboBox.ts [hunk 1]
```diff
@@ -15,6 +15,18 @@ export class CustomSelectorComboBox
...
+    protected reload(inputValue: string): Q.Promise<any> {
+        if (this.getLoader().isLoaded() && this.getLoader().getSearchString() === inputValue) {
+            return this.getLoader().search(inputValue);
+        }
+
+        this.getLoader().setSearchString(inputValue);
+
+        return this.getLoader().load().then(() => {
+            return this.getLoader().search(inputValue);
+        });
```
Comment (alansemenov) [inline modules/lib/src/main/resources/assets/js/app/inputtype/customselector/CustomSelectorComboBox.ts]: Can we replace `this.getLoader()` with a `const` please?

Comment (ashklianko) [reply modules/lib/src/main/resources/assets/js/app/inputtype/customselector/CustomSelectorComboBox.ts]: Done

**Review by alansemenov — APPROVED** [review]

**Pull request merged**